sftp/sftp.go

```diff
@@ -23,21 +24,32 @@ type FileManager interface {
 	Delete(remoteFilePath string) error
 }
 
+type Option func(impl *fileManagerImpl)
+
+func WithRetryOnIdleConnectionLost(enableRetryOnConnectionLost bool) Option {
```

Comment:
[optional] the `enableRetryOnConnectionLost` might not be needed given that the default is false, if this function is invoked then we simply set it to `true`.

Reply:
Also, we could use a comment on this method and specify that it is going to retry an operation **_only once_** if the connection is lost.

Reply:
addressed